pgmpy/continuous/discretize/base.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import division
+
+from six import with_metaclass
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+from pgmpy.factors import ContinuousFactor
+
+
+class BaseDiscretizer(with_metaclass(ABCMeta)):
+    """
+    Base class for the discretizer classes in pgmpy. The discretizer
+    classes are used to discretize a continuous random variable
+    distribution into discrete probability masses.
+
+    Parameters
+    ----------
+    factor: ContinuousFactor
+        the continuous factor representing the distribution
+        to be discretized.
+    low, high: float
+        the range over which the function will be discretized.
+    cardinality: int
+        the number of states required in the discretized output.
+
+    """
+    def __init__(self, factor, low, high, cardinality):
+        if not isinstance(factor, ContinuousFactor):
+            raise ValueError("{factor} is not a valid ContinuousFactor object."
```

Comment:
@yashu-seth Raise the error in this format: `factor expected type: ContinuousFactor. Got: ... `. We have used this format in pgmpy. Just search for it.